elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideAutoConfiguration.java

```diff
@@ -127,17 +140,34 @@ public DataSourceConfiguration getDataSourceConfiguration() {
         };
     }
 
+    @Bean
+    @RefreshScope
+    @ConditionalOnMissingBean
+    public RefreshableElide getRefreshableElide(Elide elide) {
+        return new RefreshableElide(elide);
+    }
+
+    @Bean
+    @RefreshScope
+    @ConditionalOnMissingBean
+    public QueryRunners getQueryRunners(RefreshableElide refreshableElide) {
+        return new QueryRunners(refreshableElide);
+    }
+
     /**
      * Creates the Elide instance with standard settings.
      * @param dictionary Stores the static metadata about Elide models.
      * @param dataStore The persistence store.
+     * @param transactionRegistry Global transaction registry.
      * @param settings Elide settings.
      * @return A new elide instance.
      */
     @Bean
     @ConditionalOnMissingBean
+    @Scope(SCOPE_PROTOTYPE)
```

Comment:
Should we clean up prototype scoped objects? It is recommended [here](https://docs.spring.io/spring-framework/docs/3.0.0.M3/reference/html/ch04s04.html#beans-factory-scopes-prototype)

Reply:
Good call, but I think I might fold these beans into the the Refresh versions.  I think it is confusing having both.  They are essentially private.

Reply:
I did a small refactor.  For the beans that are still declared PROTOTYPE scope, I don't believe any cleanup is required as they don't hold onto connections or sessions.  The only thing which manages connections are transactions.  We expect transactions to complete (or timeout) after which their resources should be cleaned up.  The controllers are also RefreshScope - so they should be garbage collected along with their associated references to any prototype beans.